mlir/test/Dialect/Vector/vector-transfer-flatten.mlir

```diff
@@ -102,6 +102,25 @@ func.func @transfer_read_dims_mismatch_non_zero_indices(
 
 // -----
 
+func.func @transfer_read_dims_mismatch_contiguous_non_zero_idx(
```

Comment:
```suggestion
func.func @transfer_read_dims_mismatch_non_contiguous_non_zero_indices(
```

* contiguous -> non_contiguous (probably a typo?)
* idx -> indices (for consistency with `transfer_read_dims_mismatch_non_zero_indices`)

Similar comment for `transfer_write_dims_mismatch_contiguous_non_zero_idx`

Reply:
Sorry, I don't follow why it is `non_contiguous`? I think it is writing `vector<2x2xf32>` to a contiguous memory?

Reply:
> %subview : memref<1x3x3x2xf32, strided<[40, 10, 2, 1], offset: ?>>

Note the strides - the memref is not contiguous when considering all dims. However, your point is correct:

> I think it is writing vector<2x2xf32> to a contiguous memory?

When only considering the **trailing 2 dims**, memref is indeed contiguous. IIRC, the point of this test wast to make the pattern work even if:
* overall, memref is non-contiguous, however
* the inner dims _are_ contiguous and that's sufficient.

Does it make sense? If yes, then we probably need a better name or a comment to explain 😅

Reply:
I see, it makes sense to me now! Thanks for the explanation. I found that you're improving the comments in https://github.com/llvm/llvm-project/pull/94490. Given that it will follow the naming convention, I'll leave the "comment improvement" to the PR that you're working on. And I will update the function name in this PR.